cmake/Check128BitCas.cmake

```
@@ -25,18 +25,16 @@
 
 set(CHECK_PROGRAM
     "
-    int
-    main()
+    int main(void)
     {
         __int128_t x = 0;
-        __sync_bool_compare_and_swap(&x,0,10);
-
-        return 0;
+        return __sync_bool_compare_and_swap(&x,0,10);
```

```suggestion
        return __sync_bool_compare_and_swap(&x,0,10) ? 0 : 1;
```
As the name suggests, `__sync_bool_compare_and_swap` returns a `bool`. https://gcc.gnu.org/onlinedocs/gcc-4.1.2/gcc/Atomic-Builtins.html

---

Hmm except that `bool` isn't a type in all dialects of C:  https://godbolt.org/z/vrdE116WG

---

Not sure the sample code really needs to change at all.  Pedantic C doesn't allow nothing between the braces when declaring a function.  Some compilers might warn about ignoring the return value of __sync_bool_compare_and_swap().

---

I think the main issue isn't whether it returns a `bool` but whether the return value is correct. `__sync_bool_compare_and_swap` will return a truthy value if it succeeds, which will indicate an exit failure if main returns it. That shouldn't interfere with CMake's check, but it seems risky.

---

I don't think `check_c_source_compiles` actually runs the program